app/amigo-share-request-service/amigo-share-request.service.ts

```diff
@@ -58,12 +58,16 @@ export class AmigoShareRequestService {
             }
         );
 
-        return this.http.post(
-            this.shareRequestsUrl,
-            JSON.stringify(amigoShareRequest),
+        const options = new RequestOptions(
             {
                 headers: headers
             }
+        );
```

Comment:
In all the POST and PUT calls, I swapped using raw object and instead am using an instance of `RequestOptions` to see if that would make the Content-Type header be honored. It didn't make a difference, and still sends "text/plain" 

:(